examples/terminating-gateway-tls/README.md

```diff
@@ -0,0 +1,141 @@
+# Terminating Gateways on ECS.
+
+This example demonstrates accessing non mesh destinations services from mesh tasks with the help of terminating gateways deployed as ECS tasks.
+
+There are instructions below on how to interact with this setup.
+
+![Example architecture](https://github.com/hashicorp/terraform-aws-consul-ecs/blob/main/_docs/terminating-gateway-external-server-tls.png?raw=true)
+
+## Requirements
+
+* `jq`
+* `curl`
+* Terraform >= 1.2.2
+
+## Usage
+
+### Setup
+
+Clone this repository:
+
+```console
+$ git clone https://github.com/hashicorp/terraform-aws-consul-ecs.git
+$ git checkout tags/<latest-version>
+$ cd terraform-aws-consul-ecs/examples/terminating-gateway
+```
+
+This module contains everything needed to spin up the example. The only
+requirement is to pass in the IP address of your workstation via the `lb_ingress_ip`
+variable. This is used for the security groups on the application load balancers to ensure
+only you have access to them.
+
+Determine your public IP. You can use a site like https://ifconfig.me/:
+
+```console
+$ curl ifconfig.me
+123.456.789.1%
+```
+
+Initialize Terraform:
+
+```console
+$ terraform init
+```
+
+### Terraform Apply
+
+Then apply the Terraform passing in a name and your IP:
+
+```console
+$ terraform apply \
+    -var lb_ingress_ip=123.456.789.1
+```
+
+The plan should look similar to:
+
+```shell
+Plan: 96 to add, 0 to change, 0 to destroy.
+
+Changes to Outputs:
+  + consul_server_bootstrap_token = (known after apply)
+  + consul_server_lb_address      = (known after apply)
+  + mesh_client_lb_address        = (known after apply)
+```
+
+Type `yes` to apply the changes.
+
+once the apply is complete, certs would have been uploaded to the efs volume.
+change the `variables.tf` file to point to the correct efs volume id.
+
+Add the following to the `variables.tf` file
+```variable "volumes" {
+description = "List of volumes to include in the aws_ecs_task_definition resource."
+type        = any
+default     = [
+    {
+      name =      "certs-efs"
+      host_path =  "<mount_path_inside_container>"
+      efs_volume_configuration = {
+          file_system_id = "<your_file_system_id>"
+          root_directory = "<path_inside_file_system_where_certs_are_present>"
+          iam = "ENABLED"
+        }
+    }
+]
+}
+```
+and set `tgw_certs_enabled = true` in `variables.tf` file.
```

Comment:
This part isn't needed. We need not allow users to configure these input variables.